mantidproject/mantid
# MantidAxType not JSON serializable

This PR addresses the issue of the `MantidAxType` not being JSON serializable and thus causing a stream of warnings which some of our users found confusing. The proposed solution replaces the value stored in the 'axis' variable which is a part of the creation_args dictionary that is being JSON serialized. The value of type `MantidAxType` is replaced with its value, which currently is of integer type, which is serializable. 

**To test:**

To test whether the proposed solution removes the warning:
1. In the workbench, create a bin plot, e.g. using UnitTest/ILL/D7/396940.nxs
2. Make sure that the project recovery is enabled, you may want to lower the delay between checkpoints 
3. Wait until project recovery is triggered
4. There should be no  'Unknown error occurred. Full detail: Object of type 'MantidAxType' is not JSON serializable' warning anymore.

To test whether it does not break loading back the bin plots:
5. Save the project with the bin plot open
6. Close everything and reload.
7. The bin plot should exist and look just like at the moment of saving.

Notice: If you created a plot from a workspace group with only one workspace, saving the project seems to flatten it to a workspace, thus breaking the connection of the plot to the original workspace and in such a case, the bin plot (or any plot for the matter) will not be properly created.



Fixes #31775 . 


Release note mention added.



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
